unikernel.ml

```diff
@@ -1328,6 +1328,102 @@ struct
             Middleware.http_response reqd ~title:"Error"
               ~data:"Couldn't find fields" `Bad_request)
 
+  let download_volume json_dict albatross reqd (user : User_model.user) =
+    match
+      Utils.Json.(get "block_name" json_dict, get "compression_level" json_dict)
+    with
+    | Some (`String block_name), Some (`Int compression_level) -> (
```

Comment:
here we need a different interface for albatross, and have some streaming or chunk read & write - likely with a chunked transfer-encoding.